src/tests/quantuminspire/projectq/test_backend_qx.py

```diff
@@ -257,15 +260,18 @@ def test_reset_is_cleared(self):
         self.assertTrue(self.qi_backend.clear)
 
     @patch('quantuminspire.projectq.backend_qx.get_control_count')
-    def __store_function_assert_equal(self, identity, gate, qasm, function_mock, count=0, verbose=0):
+    def __store_function_assert_equal(self, identity, gate, qasm, function_mock, count=0, nr_of_qubits=1, verbose=0):
+        self.assertLessEqual(count + nr_of_qubits, 3)
```

Comment:
It looks like `count` and `nr_of_qubits` are not actually outputs of the code under test, but specified by the author of the test. Is a test assertion then the best way to validate these parameters, since specifying invalid parameters would not be a bug in the actual code under test.

Reply:
I raise a ValueError now. now we get a unit test error.